control-plane/agents/src/bin/core/tests/volume/switchover.rs

```diff
@@ -173,6 +187,19 @@ async fn old_nexus_delete_after_vol_destroy() {
     assert_eq!(nexus_spec.len(), 0, "spec should not contain any nexus");
 }
 
+/// Checks if node is online, returns true if yes.
+async fn is_node_online(node_client: &impl NodeOperations, node_id: NodeId) -> bool {
+    let node = node_client
+        .get(Filter::Node(node_id), None)
+        .await
+        .expect("Cant get node object");
+    let node = node.0.get(0).unwrap();
+    if node.state().unwrap().status == NodeStatus::Online {
```

Comment:
probably don't want to unwrap here if we use this within a "retry" function.

Reply:
state will always be availble right? Even if node is offline. But will handle

Reply:
Handled unwrap for now. Im sure node object will be there so using expect while fetching node.